pallets/sponsorship/src/mock.rs

```diff
@@ -141,6 +149,7 @@ pub enum SponsorshipType {
 	Balances,
 	Uniques,
 	UniquesMint,
+	Compoud,
```

Comment:
I had imagined supporting batch operations differently. In my view batching (or as you named compound) is not a sponsorship type. `batch_sponsor` will be a sponsorship function with the following signature: `pub fn batch_sponsor(origin: OriginFor<T>, calls: Vec<<T as Config>::RuntimeCall>) -> DispatchResultWithPostInfo` at the same level with `sponsor_for`. This added function will allow a list of calls to be executed in a similar fashion as it's in the [utility pallet](https://github.com/paritytech/polkadot-sdk/blob/polkadot-v1.6.0/substrate/frame/utility/src/lib.rs#L201-L241). What we want to achieve is "if the selected sponsorship type (say Uniques) allows certain calls then those calls can also be submitted as a batch.